### ISSUE TITLE
Runit fails stage 1: "zoned: ioctl BLKGETZONESZ failed (Not a tty)"

### ISSUE THREAD

**Issue by foopub** [void-linux/void-packages#31275]
### System

* xuname:  Void 4.19.192_1 x86_64-musl GenuineIntel uptodate rrnFF

* package:  btrfs-progs-5.12.1_1

### Expected behavior

Boots normally to login shell. 

### Actual behavior

`zoned: ioctl BLKGETZONESZ failed (Not a tty)`

### Steps to reproduce the behavior

Use btrfs and boot. My fstab, might be relevant
```
UUID=       /       btrfs rw,relatime,ssd,space_cache,subvolid=5,subvol=/   0 1
UUID=       /boot/efi   vfat    rw,relatime             0 2
tmpfs           /tmp        tmpfs   defaults,nosuid,nodev   0 0
```

The btrfs-progs seems to have been updated from 5.9 to 5.12 yesterday. I have found no other mention of this issues even tho the btrfs-progs version was released on April 26 - this leads me to believe it's specific to the interaction between runit and btrfs-progs. 

Relevant line:

https://github.com/kdave/btrfs-progs/blob/b16b0a766f06138de2ee32d4d84b7110e469ff49/kernel-shared/zoned.c#L492

**Comment by ericonr** [comment]
And this feels like btrfs-progs broke compatibility with older kernels, could you try using a newer kernel so we can confirm?

**Comment by FollieHiyuki** [comment]
I just installed linux4.19 to try out. I got kernel panic.

The incompatibility feature is zoned mode (introduced in v5.12):

https://btrfs.wiki.kernel.org/index.php/Zoned

As stated in the link it requires new kernel to access the file system (read/write).

And after digging the changelogs (my bad for not doing this thoroughly in the PR), the license for `libbtrfsutil` changed to LGPL v2.1+ (in 5.12) and `libmount` is added to the build dependencies (since 5.10). I probably will make a PR soon to adjust these changes.

**Comment by FollieHiyuki** [comment]
Oh never mind `libmount` was dropped in v5.11 :)

**Comment by ericonr** [comment]
@FollieHiyuki right but booting with old kernels should still just work...

**Comment by FollieHiyuki** [comment]
But how can I boot my kernel when it can't read my filesystem.

linux5.4 gets kernel panic too with trace `Unable to mount fs on unknown-block`. And linux5.4 used to work on my machine.
The INSTALL file mentions 5.10+ for assuring the feature to work though. So should we revert to btrfs-progs-5.11.1 for compability?

**Comment by foopub** [comment]
> Does this failure at least dump you to a shell where you can fix things?

Yes, everything works fine after I just Ctrl+d out of the shell without doing anything. 

> And this feels like btrfs-progs broke compatibility with older kernels, could you try using a newer kernel so we can confirm?

Works find with 5.2, presumably higher versions too.

> I just installed linux4.19 to try out. I got kernel panic.

That's odd, I didn't get kernel panic.

**Comment by FollieHiyuki** [comment]
Hmm then probably it's not `btrfs-progs` that broke for me, but something else :) I simultaneously run both 5.10 and 5.11 and they run just fine though, so don't really know what the deal actually is.

**Comment by foopub** [comment]
> So should we revert to btrfs-progs-5.11.1 for compability?

I think  compiling 5.12 without the zoned module would work fine too. I can also check if I get the same problem on Arch with systemd (assuming they compile it with zoned module).

**Comment by FollieHiyuki** [comment]
Oops I feel dumb making bold move :)

I forgot I set `compress=zstd` in /etc/fstab, so of course linux4.19 will not work. And my dracut uses zstd too, which needs linux5.9 or later. Gotta go back to gzip and try again

**Comment by ericonr** [comment]
Ok, this was reproducible locally.

**Comment by ericonr** [comment]
Ok, I opened https://github.com/kdave/btrfs-progs/issues/376
